datafusion/execution/src/config.rs

```diff
@@ -198,6 +201,12 @@ impl SessionConfig {
         self
     }
 
+    /// Insert new [ConfigExtension]
+    pub fn with_option_extension<T: ConfigExtension>(mut self, extension: T) -> Self {
```

Comment:
we have discussed `with_extension_options` as the method name, but it make more sense to me to name it like this, not to be confused with `with_extension`

Reply:
Yeah I agree this is a reasonable name, even though it still a bit awkward

Reply:
I'm horrible with naming things 😀 open to suggestions

Reply:
Yeah, if I had a better alternative I would give it.